vernac/library.ml

```diff
@@ -377,18 +369,12 @@ let require_library_from_dirpath ~lib_resolver modrefl export =
   let needed, contents = List.fold_left (rec_intern_library ~lib_resolver) ([], DPmap.empty) modrefl in
   let needed = List.rev_map (fun dir -> DPmap.find dir contents) needed in
   let modrefl = List.map fst modrefl in
-  if Lib.is_module_or_modtype () then
-    begin
-      warn_require_in_module ();
-      add_anonymous_leaf (in_require (needed,modrefl,None));
-      Option.iter (fun export ->
-          (* TODO import filters *)
-          List.iter (fun m -> Declaremods.import_module unfiltered ~export (MPfile m)) modrefl)
-        export
-    end
-  else
-    add_anonymous_leaf (in_require (needed,modrefl,export));
-  ()
+  if Lib.is_module_or_modtype () then warn_require_in_module ();
+  add_anonymous_leaf (in_require (needed,modrefl));
+  Option.iter (fun export ->
+      (* TODO import filters *)
+      List.iter (fun m -> Declaremods.import_module unfiltered ~export (MPfile m)) modrefl)
```

Comment:
BTW this might be nicer using the `import_modules` and list map from open_require instead of List.iter

Reply:
Ah ah, I was reading the same line. I'll update, better resolve the duplication in the other way indeed.

Reply:
Done.